pkg/virt-controller/services/multus_annotations_test.go

```diff
@@ -86,4 +88,73 @@ var _ = Describe("Multus annotations", func() {
 			Expect(multusAnnotationPool.toString()).To(BeIdenticalTo(expectedString))
 		})
 	})
+
+	Context("Generate Multus network selection annotation", func() {
+		var vmi *v1.VirtualMachineInstance
+
+		BeforeEach(func() {
+			vmi = &v1.VirtualMachineInstance{ObjectMeta: metav1.ObjectMeta{Name: "testvmi", Namespace: "default"}}
+			vmi.Spec.Networks = []v1.Network{
+				{Name: "default", NetworkSource: v1.NetworkSource{Pod: &v1.PodNetwork{}}},
+				{Name: "blue", NetworkSource: v1.NetworkSource{Multus: &v1.MultusNetwork{NetworkName: "test1"}}},
+				{Name: "red", NetworkSource: v1.NetworkSource{Multus: &v1.MultusNetwork{NetworkName: "other-namespace/test1"}}},
+			}
+			vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{
+				{Name: "default", Binding: &v1.PluginBinding{Name: "test-binding"}},
+				{Name: "blue"}, {Name: "red"}}
+		})
+
+		When("NetworkBindingPlugins feature enabled", func() {
+			It("should fail if the specified network binding plugin is not registered (specified in Kubevirt config)", func() {
+				kvConfig := &v1.KubeVirtConfiguration{
+					DeveloperConfiguration: &v1.DeveloperConfiguration{
+						FeatureGates: []string{"NetworkBindingPlugins"}}}
+				config, _, _ := testutils.NewFakeClusterConfigUsingKVConfig(kvConfig)
+
+				_, err := GenerateMultusCNIAnnotation(vmi.Namespace, vmi.Spec.Domain.Devices.Interfaces, vmi.Spec.Networks, config)
+
+				Expect(err).To(HaveOccurred())
+			})
+
+			It("should add network binding plugin net-attach-def to multus annotation", func() {
+				kvConfig := &v1.KubeVirtConfiguration{
+					DeveloperConfiguration: &v1.DeveloperConfiguration{
+						FeatureGates: []string{"NetworkBindingPlugins"}},
+					NetworkConfiguration: &v1.NetworkConfiguration{
+						Binding: map[string]v1.InterfaceBindingPlugin{
+							"test-binding": {NetworkAttachmentDefinition: "test-binding-net"}}}}
+				config, _, _ := testutils.NewFakeClusterConfigUsingKVConfig(kvConfig)
+
+				Expect(GenerateMultusCNIAnnotation(vmi.Namespace, vmi.Spec.Domain.Devices.Interfaces, vmi.Spec.Networks, config)).To(MatchJSON(
+					`[
+						{"name": "test-binding-net","namespace": "default", "cni-args": {"name": "eth0"}},						
+						{"name": "test1","namespace": "default","interface": "pod16477688c0e"},
+						{"name": "test1","namespace": "other-namespace","interface": "podb1f51a511f1"}
+					]`,
+				))
+			})
+
+			DescribeTable("should parse NetworkAttachmentDefinition name and namespace correctly, given",
+				func(netAttachDefRawName, expectedAnnot string) {
+					vmi.Spec.Networks = []v1.Network{*v1.DefaultPodNetwork()}
```

Comment:
I do not think we should use a common setup and then in one incident override it completely.
The common setup should consist of the common things, then you can modify it in each test.
If there is no common setup, then pass it to each test or use helpers to call them from the tests.

Reply:
Done